### PR TITLE
EWS - Entry cotext standarts for Email

### PR DESCRIPTION
**Pull request by GalRabin**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/21835

## Description
Adding email entry context to `ews-get-items` command, So customers can refer to email in global entry context path.

## Screenshots
Example of entry context of `ews-get-items` command:
image

## Does it break backward compatibility?
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by GalRabin** [comment]
@idovandijk 
Can you confirm that this answer the request in the issue, I will share with YAML for evaluation, Thanks

**Comment by idovandijk** [comment]
> @idovandijk
> Can you confirm that this answer the request in the issue, I will share with YAML for evaluation, Thanks
> 
> Before writing tests and documentation

perfect

**Comment by GalRabin** [comment]
I will add docs and I will merge next week

File: Packs/Legacy/Integrations/EWSv2/EWSv2.py [hunk 1]
```diff
@@ -1691,7 +1704,8 @@ def get_items(item_ids, target_mailbox=None):
         'ReadableContentsFormat': formats['markdown'],
         'HumanReadable': tableToMarkdown('Get items', items_to_context, ITEMS_RESULTS_HEADERS),
         ENTRY_CONTEXT: {
-            CONTEXT_UPDATE_EWS_ITEM: items_to_context
+            CONTEXT_UPDATE_EWS_ITEM: items_to_context,
+            'Email': [email_ec(item) for item in items]
```
Comment (yaakovi) [inline Packs/Legacy/Integrations/EWSv2/EWSv2.py:1708]:
```suggestion
            'Email': [email_ec(item) for item in items],
```

File: Packs/Legacy/Integrations/EWSv2/EWSv2.py [hunk 2]
```diff
@@ -946,6 +946,19 @@ def str_to_camel_case(snake_str):
     return str_to_camel_case(value)
 
 
+def email_ec(item):
+    return {
+        'CC': [] if not item.cc_recipients else [mailbox.email_address for mailbox in item.cc_recipients],
+        'BCC': [] if not item.bcc_recipients else [mailbox.email_address for mailbox in item.bcc_recipients],
+        'To': [] if not item.to_recipients else [mailbox.email_address for mailbox in item.to_recipients],
```
Comment (yaakovi) [inline Packs/Legacy/Integrations/EWSv2/EWSv2.py:953]: 1) are we sure we want an empty list as a default value and not omitting the key from the object?
2) when empty, what `item.*_ recipients` contain? empty list or None? if list then the condition is redundant.

File: Packs/Legacy/Integrations/EWSv2/EWSv2.py [hunk 3]
```diff
@@ -946,6 +946,19 @@ def str_to_camel_case(snake_str):
...
+        'From': item.author.email_address,
+        'Subject': item.subject,
+        'Text': item.text_body,
+        'HTML': item.body,
+        'HeadersMap': {header.name: header.value for header in item.headers}
```
Comment (yaakovi) [inline Packs/Legacy/Integrations/EWSv2/EWSv2.py:958]:
```suggestion
        'HeadersMap': {header.name: header.value for header in item.headers},
```

File: Packs/Legacy/Integrations/EWSv2/EWSv2.py (hunk 1 above)
Comment (GalRabin) [reply Packs/Legacy/Integrations/EWSv2/EWSv2.py:1708]: Same

File: Packs/Legacy/Integrations/EWSv2/EWSv2.py (hunk 3 above)
Comment (GalRabin) [reply Packs/Legacy/Integrations/EWSv2/EWSv2.py:958]: Why we need to add comma in last item of dict or item?

File: Packs/Legacy/Integrations/EWSv2/EWSv2.py (hunk 2 above)
Comment (GalRabin) [reply Packs/Legacy/Integrations/EWSv2/EWSv2.py:953]: Changed to None

**Comment by GalRabin** [comment]
@kirbles19 
Hi,
I will appriciate your review on:
1. [Readme.md](https://github.com/demisto/content/blob/ews-get-items/Packs/EWS/Integrations/EWSv2/README.md) - Lines 2009 - 2047 
2. [YAML](https://github.com/demisto/content/blob/ews-get-items/Packs/EWS/Integrations/EWSv2/EWSv2.yml) - Lines 883-906

**Comment by kirbles19** [comment]
@GalRabin Looks good.

**Pull request merged**